.github/workflows/build-soroban-dev.yml

```diff
@@ -42,3 +44,32 @@ jobs:
           "network": ["standalone"],
           "options": ["", "--enable-horizon-captive-core", "--enable-soroban-rpc"],
         }
+
+  arm64:
+    uses: ./.github/workflows/build.yml
+    secrets:
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+    with:
+      arch: arm64
+      tag: soroban-dev-arm64
+      core_ref: c0ad35aa19297e112d71fcc5755458495f99a237
+      core_configure_flags: --disable-tests --enable-next-protocol-version-unsafe-for-production
+      core_build_runner_type: ubuntu-latest-16-cores
+      go_ref: soroban-v0.0.4
+      soroban_tools_ref: v0.4.0
+      test_matrix: |
+        {
+          "network": ["standalone"],
+          "options": ["", "--enable-horizon-captive-core", "--enable-soroban-rpc"],
+        }
+
+  manifest:
```

Comment:
is multi-platform usage such as `buildx build --platform linux/arm64,linux/amd64` not possible in this context, if it was, then could avoid external manifest and separate `image_name-<platform>`, rather buildx would generate single image with platform's bundled within? at least that's how [this article showed doing it](https://blog.container-solutions.com/building-multiplatform-container-images) 

I haven't done that multi-format before, and not sure if it's viable in this copy from multi-image's to separate final image build assembly, but if were, could reduce boilerplate here and from a user experience, would only need to reference `image_name` and the platform gets pulled dynamically by docker based on host arch.

Reply:
That's a good question. `buildx` uses buildkit and container backends instead of the original docker backend which is really cool if the features you need are supported by it. I've found buildkit will generate smaller images for some reason, and yeah it will even do multi-image builds for multiple archs at the same time which is really cool.

We actually use buildx to build the intermediary images, but I'm not using it to build the final quickstart image for a few reasons:

1. Buildx doesn't support loading images from files. The build process currently builds images for stellar-core, horizon, friendbot, soroban-rpc, and then copies files out of them when building the quickstart image. To do that it has to load those original images, however buildkit when using a container backend is only able to get images from a registry.
1. Buildx doesn't support exporting to a docker image file when building for multiple platforms at the same time. This makes it quite difficult to test an image offline before pushing it to a registry. Quite a few people have raised this issue already on the Docker issue tracker for years, but unfortunately with the newer buildkit stuff it just doesn't seem to be a priority to develop features that don't use registries.
1. Multi-image images are actually a manifest list that references other manifests, and manifest lists can only exist on registries. So the only way to test outside of a registry conveniently is really to have two images anyway.
1. Building both images at once would mean building them on a single GitHub Runner. As you might have noticed with the build there's a lot of building we can do in parallel and we'd basically lose that. Additionally when GitHub releases ARM64 runners we'll be able to build the ARM64 image natively and it'll be much faster, so again there's a benefit to building separately.

For the first two issues we could try running a registry inside each workflow job. We'd have to somehow save the state of the registry and reload it in each job. To run the jobs concurrently we'd actually need four registries. It all seemed pretty complicated so I don't think it is worth it.

Happy to chat about this more. Buildx is great if it's limited capabilities just happen to fit what we're needing to do, but the moment you step outside that it's much easier to use the regular docker commands.

Reply:
One more thing, I have some thoughts on refactoring the build process so we don't actually need to load the intermediary images anymore. I'd be interested in exploring two alternative approaches sometime:
1. Extracting the binaries out of the images outside of docker, and then just adding them as regular files. This would eliminate the need to load images into the docker runtime.
2. Adding the intermediary Dockerfiles as stages in the quickstart image. This would make the entire build process simpler, although it would also make it more serial, so it comes with a pretty big disadvantage I think.

I'm most likely to experiment with (1) I think, but low priority at the moment.

Reply:
yeah, sounds good. option#1 may be error prone, handling files that are platform specific, i.e. you can't copy the arm64 bin into amd64 image. something like option#2 may allow for the single buildx invocation with muilti-platform, thus eliminating need for external manifest management and multiple image names permutated on platform, any path that results in concatenating down to a single dockerfile could then be eligible as single buildx with multi-platform.

Reply:
Even if we did multi-stage, we'd still need to manage a registry to be able to test the image locally, that registry state would be persisted across jobs, and that single job would run on a single instance architecture making arm64 builds slow after GitHub releases arm64 runners. So I doubt we'd use a single buildx command, there's so many downsides.